snakes/python/summer-league-2022/src/logic.py

```diff
@@ -104,13 +103,39 @@ def _avoid_my_body(my_body: dict, possible_moves: List[str]) -> List[str]:
 
     my_head = my_body[0]
 
-    if my_head["x"] + 1 in [body["x"] for body in my_body[1:]]:
+    for segment in my_body:
+        if segment["x"] == my_head["x"] and segment["y"] == my_head["y"]:
+            continue
+        if segment["x"] < my_head["x"]:
+            if "left" in possible_moves:
+                possible_moves.remove("left")
+        elif segment["x"] > my_head["x"]:
+            if "right" in possible_moves:
+                possible_moves.remove("right")
+        elif segment["y"] < my_head["y"]:
+            if "down" in possible_moves:
+                possible_moves.remove("down")
+        elif segment["y"] > my_head["y"]:
+            if "up" in possible_moves:
+                possible_moves.remove("up")
+
+    return possible_moves
```

Comment:
So I (ahem... Copilot) kinda rewrote this a bit. I think I can revert this part, but it does seem to pass tests also...